examples/dreamcast/filesystem/browse/browse.c

```diff
@@ -0,0 +1,379 @@
+    if(sd_blockdev_for_partition(0, &sd_dev, &partition_type)) {
+        fprintf(stderr, "Could not find the first partition on the SD card!\n");
+        return 0;
+    }
+
+    if(fs_fat_mount("/sd", &sd_dev, FS_FAT_MOUNT_READWRITE)) {
+        fprintf(stderr, "Could not mount SD card as fatfs. Please make sure the card "
+            "has been properly formatted.\n");
+        return 0;
+    }
+
+    printf("SD Card mounted Successfully\n");
+
+    return 1;
+}
+
+static void unmount_sd_fat() {
+    fs_fat_unmount("/sd");
+    fs_fat_shutdown();
+    sd_shutdown();
+}
+
+static int mount_romdisk(char *filepath, char *mountpoint) {
+    void *buffer = NULL;
+    ssize_t filesize = fs_load(filepath, &buffer);
+
+    if(filesize != -1) {
+        fs_romdisk_mount(mountpoint, buffer, 1);
+        return 1;
+    }
+    else
+        return 0;
+}
+
+static void show_prompt(char *current_directory, bool mounted_sd, bool highlight_yes) {
+    if(strcmp(current_directory, "/sd") == 0 ||
+       strcmp(current_directory, "/ram") == 0)  {
+            if(mounted_sd)
+                prompt_message("Delete this file from SD card?", highlight_yes);
+            else
+                prompt_message("Delete this file from /ram directory?", highlight_yes);
+    }
+    else {
+        if(mounted_sd)
+            prompt_message("Copy this file to SD card?", highlight_yes);
+        else 
+            prompt_message("Copy this file to /ram directory?", highlight_yes);
+    }
+}
+
+static void delete_file(char *filename, bool mounted_sd) {
+    void *filedata = NULL;
+    unsigned int filesize = 0;
+    
+    if(mounted_sd)
+        remove(filename);
+    else {
+        fs_ramdisk_detach(basename(filename), &filedata, &filesize);
+        free(filedata);
+    }
+}
+
+static int browse_directory(char *directory, directory_file_t directory_contents[]) {
+    int count = 0;
+    DIR *d;
+    struct dirent *entry;
+
+    /* Open the directory */
+    if(!(d = opendir(directory)));
+
+    /* Clear out all files */
+    memset(directory_contents, 0, sizeof(directory_contents[0])*100);
+
+    /* Read all the filenames in the directory */
+    while((entry = readdir(d)) && count < 100) {
+        directory_contents[count].is_dir = (entry->d_type == DT_DIR);
+        strncpy(directory_contents[count].filename, entry->d_name, strlen(entry->d_name));
+        count++;
+    }
+
+    /* Close directory */
+    closedir(d);
+
+    return count;
+}
+
+static void prompt_message(char *message, bool highlight_yes) {
+    int x = 20 + 24, y = 350;
+    int color = 1;
+
+    bfont_set_foreground_color(0xFFFFFFFF);
+    bfont_set_background_color(0x00000000);
+    bfont_draw_str(vram_s + y*640+x, 640, color, message);
+
+    if(highlight_yes) {
+        bfont_set_foreground_color(0x00000000);
+        bfont_set_background_color(0xFFFFFFFF);
+    }
+    else {
+        bfont_set_foreground_color(0xFFFFFFFF);
+        bfont_set_background_color(0x00000000);
+    }
+    y += 24;
```

Comment:
```suggestion
    y += BFONT_HEIGHT;
```
Maybe you'd rather set as a local define (`LINE_STRIDE` or something?) if the name doesn't seem right, but it seems quite a bit in many uses of fixed numbers you could use the defines from bfont as they reflect the bfont height/width.